app/models/api_entreprise_token.rb

```diff
@@ -6,11 +6,15 @@ def initialize(token)
   end
 
   def roles
-    decoded_token["roles"] if token.present?
+    return [] if token.blank?
+
+    Array(decoded_token["roles"])
   end
 
   def expired?
-    Time.zone.now.to_i >= decoded_token["exp"] if token.present?
+    return false if token.blank? || !decoded_token.key?("exp")
+
+    Time.zone.now.to_i >= decoded_token["exp"]
```

Comment:
Si on factorise la gestion des `blank` dans l'initializer

```suggestion
    decoded_token.key?("exp") && decoded_token["exp"] < Time.zone.now.to_i
```